#goal
---
#result
bool success
string result_msg
---
#feedback